File: app/src/main/java/com/akameko/testforupstarts/dagger/RepositoryModule.java
package com.akameko.testforupstarts.dagger;

import com.akameko.testforupstarts.repository.retrofit.Repository;

import dagger.Module;
import dagger.Provides;

@Module
public class RepositoryModule {
    @Provides
    Repository provideRepository() {
        return new Repository();
    }
}

File: app/src/main/java/com/akameko/testforupstarts/dagger/RepositoryModule.kt
package com.akameko.testforupstarts.dagger

import com.akameko.testforupstarts.repository.retrofit.Repository
import dagger.Module
import dagger.Provides

@Module
class RepositoryModule {
    @Provides
    fun provideRepository(): Repository {
        return Repository()
    }
}